operator/src/main/java/oracle/kubernetes/operator/Main.java

```diff
@@ -858,24 +866,63 @@ public NextAction onFailure(Packet packet, CallResponse<V1NamespaceList> callRes
     @Override
     protected NextAction onFailureNoRetry(Packet packet, CallResponse<V1NamespaceList> callResponse) {
       return isNotAuthorizedOrForbidden(callResponse)
-          ? doNext(packet) : super.onFailureNoRetry(packet, callResponse);
+          ? doNext(createDomainCrdAndStartNamespaces(targetNamespaces), packet) : 
+            super.onFailureNoRetry(packet, callResponse);
     }
 
     @Override
     public NextAction onSuccess(Packet packet, CallResponse<V1NamespaceList> callResponse) {
       V1NamespaceList result = callResponse.getResult();
       // don't bother processing pre-existing events
-
-      if (namespaceWatcher == null) {
-        namespaceWatcher = createNamespaceWatcher(getInitialResourceVersion(result));
+      String intialResourceVersion = getInitialResourceVersion(result);
+      List<String> nsList = getExistingNamespaces(result);
+      
+      Set<String> namespacesToStart = new TreeSet<>(targetNamespaces);
+      for (String ns : targetNamespaces) {
+        if (!nsList.contains(ns)) {
+          namespacesToStart.remove(ns);
+        }
```

Comment:
recheckDomains method also has similar filtering logic. Can yon double check and see if the logic there takes care of deleted namespaces correctly as well?

Reply:
My understanding is that recheckDomains runs periodically and it takes care of non-existing namespaces except in full recheck case. It also checks for namespaces that are removed from operator's targetNamespaces list or that are deleted from k8s cluster and stops those namespaces. In case of full recheck (which happens every 2 min), it'll check for all namespaces in targetNamespaces (including non-existing ns). However, I believe that doesn't cause liveness probe failure since it's executed in a thread separate from main thread.

Reply:
Shall we add a log (warning?) message when we detect that a namespace that is in the target namespace but does not exist?
We may also want to add test cases for different lifecycle sequences of namespaces w.r.t targetNamesapces. For example, a namespace that is in the targetNamespaces but does not exist initially, but is added while the operator is running or not running (and later starts to run). There are multiple permutations to be tested.  If we agree that those test cases are needed, we need to create a task to track it.

Reply:
Yes, I have added a warning message when we detect that a namespace that is in target namespace but doesn't exist. Adding test cases for different lifecycle sequences of namespaces is a good idea. I'll check on that. Thanks.